### ISSUE TITLE
Introspecting complex default values raises Protocol.UndefinedError

### ISSUE THREAD

**Issue by mydearxym** [absinthe-graphql/absinthe#490]
hi guys ~

i am using `default_value` in `input_object` ,  it *worked* , but there some error in console when i refresh the schema in graphql-playground ..

```elixir
input_object :article_filter do
    field(:first, :integer, default_value: 20)
    field(:order, :sort_order, default_value: :desc)
    field(:tag, :string, default_value: :all)
  end

field :posts, non_null(list_of(non_null(:post))) do
    arg(:filter, :article_filter, default_value: %{first: 20})
    resolve(&Resolvers.CMS.posts/3)
end
```

![image](https://user-images.githubusercontent.com/6184465/35100832-31fb6fc6-fc98-11e7-995a-8f631cbb28fa.png)

![image](https://user-images.githubusercontent.com/6184465/35100853-45e912f4-fc98-11e7-9bbb-d686afcbe2b5.png)

error detail:
```elixir
Request: POST /graphiql
** (exit) an exception was raised:
    ** (Protocol.UndefinedError) protocol String.Chars not implemented for %{first: 20}. This protocol is implemented for: Atom, BitString, Date, DateTime, Decimal, Ecto.Date, Ecto.DateTime, Ecto.Time, Float, Integer, List, NaiveDateTime, Postgrex.Copy, Postgrex.Query, Postgrex.Stream, Time, URI, Version, Version.Requirement

```

**Comment by bruce** [comment]
This does appear to be a bug, based on my reading of http://facebook.github.io/graphql/October2016/#sec-The-__InputValue-Type

"defaultValue may return a String encoding (using the GraphQL language) of the default value used by this input value in the condition a value is not provided at runtime. If this input value has no default value, returns null."

I'd take a wild bet we haven't seen this before because this is the first complex default value someone's tried to introspect.

**Comment by binaryseed** [comment]
We also ran into this recently...

**Comment by richeterre** [comment]
@bruce @binaryseed  Has this fix been released yet? I'm running `v1.4.16` and see a similar error when providing an empty map as the default value for my field arg:

```elixir
field :performances, list_of(non_null(:performance)) do
  arg :filter, :performance_filter, default_value: %{}
  ...
end

input_object :performance_filter do
  field :stage_date, :date
  field :stage_id, :id
end
```

Reloading the schema in GraphQL Playground causes the following console output:

```
** (exit) an exception was raised:
    ** (Protocol.UndefinedError) protocol String.Chars not implemented for %{}. This protocol is implemented for: Postgrex.Copy, Postgrex.Query, Decimal, Float, DateTime, Time, List, Version.Requirement, Atom, Integer, Version, Date, BitString, NaiveDateTime, URI
```

Am I missing anything here? Any pointers are greatly appreciated 🙂

**Comment by benwilson512** [comment]
@richeterre this fix is in the 1.5 series of releases, we may not have back ported it to 1.4
